daemon/algod/api/server/v2/dryrun.go

```diff
@@ -397,16 +397,19 @@ func doDryrunRequest(dr *DryrunRequest, response *generated.DryrunResponse) {
 		return
 	}
 	proto := config.Consensus[protocol.ConsensusVersion(dr.ProtocolVersion)]
-
+	proto.EnableAppCostPooling = true
```

Comment:
why is it set here? It is up to the caller or current protocol.

Reply:
The idea here is to use the fee pooling mechanism, regardless of whether it is turned on in the protocol, for two reasons:
a) It gives us a way to access the cost (because it will be decremented if it's active)
b) It gives us a way to overrun the allowable cost, so we can accurately report overruns (within reason), which seems useful to developers.  We ought to indicate that the execution "failed" but we can show the real cost.